src/chat/chat_panel.rs

```diff
@@ -929,6 +869,86 @@ impl ChatPanel {
             auto_scroll_cancellable: false,
         };
     }
+
+    fn update_visibilities(&mut self, scope: &mut Scope) {
+        macro_rules! show {
```

Comment:
This tiny detail makes the imperative visibility code shorter inside this function :) A simple closure would be a bit longer because it would need to receive an `id!(...)` as an argument, also, it may cause issues because it would need to capture mutable references to `self`.

Reply:
not a big fan of this. im not against macros, but in this case it doesnt make much sense imo. its called `hide`, but it only hides views inside self, its not clear what it recieves, since the id becomes just a token that only makes sense when looking at the macro using it in onether macro (`id!`). leaving it exist only inside of this only function make it less bad, but still, seems like complexity without any real value inside of it, that is only use in this part of the project.

Reply:
I have no problem on changing this if someone else agrees 👍 I just wanted the code to look more clean and shorter by not needing to write `self.view(id!(SOMETHING)).set_visible(SOMETHING)` all over the place. But I know it functionally makes no difference.  I would like to know what someone else thinks before removing this 🙂

Anyways, clarifying specific parts of what you said:

> but it only hides views inside self

This is why it's defined just here and not as some kind of global utility.

>  its not clear what it recieves, since the id becomes just a token that only makes sense when looking at the macro

I don't know if I'm understanding this part correctly so please excuse me if my answer to this is not precise. But as far as I understand, this would happen with a closure as well but that is not a problem because it exists inside this function and you have the implementation visible.

Reply:
you cannot just pass a bare token identifier to a clousure, you'd need to pass a &[LiveId], which is what id! gives u, though this is a less important detail.

Reply:
i would just bind the views at the beginning and then do set_visible()

```rust
let empty_conversation_view = self.view(id!(empty_conversation)); 
...
match state {
    ...
    empty_conversation_view.set_visible(false)
```

and so on.

Reply:
> you cannot just pass a bare token identifier to a clousure

Yes, that's why it's a macro and not a function. Thanks to that you can do `show!(something)` and not `show(id!(something)`.

> i would just bind the views at the beginning and then do set_visible()

Sounds good enough to me. Will change it to something like that then 👍

Reply:
I know this is resolved but adding to the conversation

+1 to the idea of avoiding unnecessary local macros. I like the idea of having a shortcut for visibility but I would explore it within Makepad itself (I'd like `hide` and `show` methods) or as a separate helper crate, so that we make it consistent throughout the app and our other apps as well.

I'd also add a +1 to the notion of making clear what `something` is in `show!(something)`. Also almost everything in Makepad that deals with `LiveId` looks like `animator_play(id!(hover.on));` or `self.view(id!(some_view))`, I think I'd be best to be consistent with that (or rethink it within Makepad).